GCRCatalogs/register.py

```diff
@@ -12,6 +13,21 @@
 _GITHUB_URL = 'https://raw.githubusercontent.com/LSSTDESC/gcr-catalogs/master/GCRCatalogs'
 _YAML_EXTENSIONS = ('.yaml', '.yml')
 
+# Keys appearing in yaml files whose values may be paths relative to _ROOT_DIR
+_PATH_LIKE = ('filename', 'addon_filename', 'base_dir', 'root_dir',
+              'catalog_root_dir', 'header_file', 'repo')
+
+def _get_rootdir():
+    with open(os.path.join(os.path.dirname(__file__),'site_config/site_rootdir.yaml')) as f:
+        d = yaml.safe_load(f)
+        host = socket.getfqdn()
+        for (k,v) in d.items():
+            if k in host:
+                return v
+        return None
```

Comment:
Make `_get_host_info` a separate function (in case in the future we want to change how to obtain host info)

If the host cannot be identified, use current directory as `ROOT_DIR`. 

```suggestion
def _get_host_info():
	return socket.getfqdn()

def _get_root_dir():
	host = _get_host_info()
    with open(os.path.join(os.path.dirname(__file__), 'site_config', 'site_rootdir.yaml')) as f:
        d = yaml.safe_load(f)
    for k, v in d.items():
        if k in host:
            return v
    return os.curdir
```

Reply:
Yes, originally I had it as a separate function.  I'll put it back.

I don't think there is a sensible fallback if host cannot be identified.  There should be a way for the user to explicitly set it, though, and an exception raised if someone tries a load when it isn't set (or at least when they try a load of something involving a relative path).

Reply:
We can raise a warning if `_DEFAULT_ROOT_DIR` cannot be set but it should not raise an exception. People may want to use GCRCatalogs in ways that do not require `ROOT_DIR`.

Reply:
If the path is absolute ROOT_DIR isn't needed.   Otherwise, I still think there is no sensible default. People may have been in the habit of depending on relative paths this way; that doesn't mean we have to continue allowing it. Having to make an extra call to set ROOT_DIR is not more trouble than having to cd to a particular place, and there is at least some record of what was done.

Reply:
I am not sure if I follow. My point is that one should be able to import `GCRCatalogs` outside CC/NERSC and without the need to set `ROOT_DIR`. For example, a user may want to just load a reader for other purposes, and there is no need for the config files at all. 

So my request is that if there's no `ROOT_DIR` at all, `GCRCatalogs` should not raise an exception (but a warning is ok).

Reply:
I was thinking of raising an exception only when someone tried to load a file without absolute path, not at import time.   If we go with explicitly including $ROOT_DIR in paths, then it would only be when loading a file for which $ROOT_DIR is part of the path.

I'd also be in favor of a warning at import time.

Reply:
At catalog load time, usually the reader does raise an exception if there is no file at the given path. 

The code we are working here loads the config, modify it as needed, and then pass it to the reader (and the reader loads the catalog). So the question is whether an exception is needed during this process. Once the config is passed to the reader, I would prefer that we let each reader to handle what exception to raise.

Reply:
For individual testing and debugging the approach of waiting to raise an exception until it's absolutely necessary has its place, though in this case I would argue that asking the user to either fix the config or make one function call to establish an acceptable root_dir is not much of a burden.   For production running - and GCRCatalogs is used extensively in such circumstances - it's better to find out about potential problems early.

Reply:
I fully agree with the distinction about interactive vs. production use. I would argue that `GCRCatalogs` is meant for interactive use, but I also have to agree that it has been used in production too.

But I am still not sure why `set_root_dir` need to raise exception. I think for production one won't need to use `set_root_dir`. If `set_root_dir` will raise exception when the path is not valid, should `config_overwrite` also raise exception when people overwrite the path with a invalid path? I would not want to go down that route.

Reply:
I agree set_root_dir is not intended for production use and it's very unlikely it would be used in those circumstances, so a warning is adequate.  I'll make this change.

config_overwrite should do the same level of checking as is done for the original path, whatever that level is. 
 I would prefer to do whatever checking can be done easily in a uniform way in register rather than duplicating the logic in many readers.  That would include looking at paths (if an invalid path has been specified, chances are good it's a typo), but, for this project, I don't think it's appropriate to make wholesale changes, especially when we have philosophical differences.

Reply:
Concerning config_overwrite: this strikes me as a place where it *is* reasonable for someone to specify an absolute path or a relative path which is relative to current dir rather than the root dir.   But it's also possible they'd want to specify something relative to root dir. This is another argument for starting the value of `base_dir` (or values of other path-like keys) with a special string when it's meant to be relative to root dir.

Reply:
I'll reply to this below and resolving this thread since the original code is outdated.